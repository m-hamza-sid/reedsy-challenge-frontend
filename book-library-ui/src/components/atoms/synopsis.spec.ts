import { mount } from "@vue/test-utils";
import { describe, expect, test } from "vitest";
import Synopsis from "./synopsis.vue";

describe("author", () => {
  test("should render text without truncation if limit is not given", () => {
    const iSynopsis =
      "In Search of Lost Time, also translated as Remembrance of Things Past, novel in seven parts by Marcel Proust, " +
      "published in French as À la recherche du temps perdu from 1913 to 1927. The novel is the story of Proust's own" +
      " life, told as an allegorical search for truth.\nIn Search of Lost Time, also translated as Remembrance of " +
      "Things Past, novel in seven parts by Marcel Proust, published in French as À la recherche du temps perdu from " +
      "1913 to 1927. The novel is the story of Proust's own life, told as an allegorical search for truth.";

    const wrapper = mount(Synopsis, { props: { synopsis: iSynopsis } });

    expect(wrapper.text().length).toEqual(iSynopsis.length);
  });

  test("should render text with truncation as per given limit and ellipses", () => {
    const iLimit = 30;
    const iSynopsis =
      "In Search of Lost Time, also translated as Remembrance of Things Past, novel in seven parts by Marcel Proust, " +
      "published in French as À la recherche du temps perdu from 1913 to 1927. The novel is the story of Proust's own" +
      " life, told as an allegorical search for truth.\nIn Search of Lost Time, also translated as Remembrance of " +
      "Things Past, novel in seven parts by Marcel Proust, published in French as À la recherche du temps perdu from " +
      "1913 to 1927. The novel is the story of Proust's own life, told as an allegorical search for truth.";

    const wrapper = mount(Synopsis, { props: { synopsis: iSynopsis, limit: iLimit } });

    expect(wrapper.text().length).toEqual(iLimit + 3);
  });
});
