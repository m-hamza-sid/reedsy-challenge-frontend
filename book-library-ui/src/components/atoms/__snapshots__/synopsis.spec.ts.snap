// Vitest Snapshot v1

exports[`author > should render with name prop 1`] = `"<div class=\\"text-left\\">some name</div>"`;
